src/Image.php

```diff
@@ -40,6 +44,17 @@ public function __construct($record = null, $isSingleton = false, $queryParams =
         $this->File->setAllowedCategories('image/supported');
     }
 
+    public function scaffoldFormFieldForHasOne(
+        string $fieldName,
+        ?string $fieldTitle,
+        string $relationName,
+        DataObject $ownerRecord
+    ): FormField&FileHandleField {
+        $field = parent::scaffoldFormFieldForHasOne($fieldName, $fieldTitle, $relationName, $ownerRecord);
+        $field->setAllowedFileCategories('image/supported');
```

Comment:
```suggestion
        if ($field->hasMethod('setAllowedFileCategories')) {
            $field->setAllowedFileCategories('image/supported');
        }
```
Might as well do the same check as above since the FileHandleField could be changed with Injector

Reply:
`setAllowedFileCategories()` is defined in the [`FileHandleField`](https://github.com/silverstripe/silverstripe-framework/blob/5/src/Forms/FileHandleField.php) interface, so we don't need to check for it.

`setAllowedMaxFileNumber()` isn't defined in that interface, which is why the condition for that method is there.